pkg/endpoint/endpoint.go

```diff
@@ -128,6 +128,11 @@ type Endpoint struct {
 	// recalculated on endpoint restore.
 	createdAt time.Time
 
+	// lastUpdatedTime store the time the endpoint was last changed.
+	// it is used for calculate endpoint progation delay. and we using
+	// a map not a single value for concurrent operation consideration.
```

Comment:
How about:

```suggestion
	// lastUpdatedTime stores the time the endpoint was last changed.
	// It is used to calculate the endpoint propagation delay.
	// We use a map, rather than a single value, to account for concurrent operations.
```